### PR TITLE
refactor(ui): add repository select & blob header of source code browser

### PR DESCRIPTION
**Pull request by liangfung**
https://jam.dev/c/580420e4-b12d-422c-8722-64b5791c2394

fix TAB-539

### CODE REVIEW

File: ee/tabby-ui/components/ui/select.tsx [hunk 1]
```diff
@@ -75,8 +75,10 @@ SelectLabel.displayName = SelectPrimitive.Label.displayName
 
 const SelectItem = React.forwardRef<
   React.ElementRef<typeof SelectPrimitive.Item>,
-  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item>
->(({ className, children, ...props }, ref) => (
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item> & {
+    emptyItem?: boolean
```
Comment (wsxiaoys) [inline ee/tabby-ui/components/ui/select.tsx:79]:
```suggestion
   placeholder: boolean
```

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

**Comment by wsxiaoys** [comment]
Please also rebase against main

File: ee/tabby-ui/app/globals.css [hunk 2]
```diff
@@ -60,7 +60,7 @@
     --primary: 39 32% 87%;
     --primary-foreground: 39 3.2% 8.7%;
 
-    --secondary: 39 16% 26.1%;
+    --secondary: 39 13% 21%;
```
Comment (liangfung) [inline ee/tabby-ui/app/globals.css:63]: Updated the secondary color to differentiate it from the border color

File: ee/tabby-ui/components/ui/select.tsx [hunk 3]
```diff
@@ -75,8 +75,10 @@ SelectLabel.displayName = SelectPrimitive.Label.displayName
 
 const SelectItem = React.forwardRef<
   React.ElementRef<typeof SelectPrimitive.Item>,
-  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item>
->(({ className, children, ...props }, ref) => (
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item> & {
+    isPlaceHolder?: boolean
```
Comment (liangfung) [inline ee/tabby-ui/components/ui/select.tsx:79]: The ‘placeholder’ attribute already exists, rename it.

**Review by wsxiaoys — APPROVED** [review]

**Pull request merged**